tests/phpunit/TestFramework/PhpUnit/CommandLine/ArgumentsAndOptionsBuilderTest.php

```diff
@@ -151,25 +151,50 @@ public function provideTestCases(): Generator
                 'App\Test::test_case1',
                 'App\Test::test_case2',
             ],
-            '/App\\\\Test\:\:test_case1|App\\\\Test\:\:test_case2/',
+            '/Test\:\:test_case1|Test\:\:test_case2/',
         ];
 
-        yield '2 simple test cases, 1 with data set and special character >' => [
+        yield '1 simple test case, 1 with data set and special character >' => [
             true,
             [
                 'App\Test::test_case1 with data set "With special character >"',
                 'App\Test::test_case2',
             ],
-            '/App\\\\Test\:\:test_case1 with data set "With special character \\>"|App\\\\Test\:\:test_case2/',
+            '/Test\:\:test_case1 with data set "With special character \\>"|Test\:\:test_case2/',
         ];
 
-        yield '2 simple test cases, 1 with data set and special character @' => [
+        yield '1 simple test case, 1 with data set and special character @' => [
             true,
             [
                 'App\Test::test_case1 with data set "With special character @"',
                 'App\Test::test_case2',
             ],
-            '/App\\\\Test\:\:test_case1 with data set "With special character @"|App\\\\Test\:\:test_case2/',
+            '/Test\:\:test_case1 with data set "With special character @"|Test\:\:test_case2/',
+        ];
+
+        yield '2 data sets from data provider for the same test case' => [
+            true,
+            [
+                'App\Test::test_case1 with data set "#1"',
+                'App\Test::test_case1 with data set "#2"',
+            ],
+            '/Test\:\:test_case1 with data set "\#1"|Test\:\:test_case1 with data set "\#2"/',
```

Comment:
do I get it right, that we are running

`phpunit --filter='Test::test_case1 with data set "With special character @"'`

?

if so, I think we can shorten this even more. PHPUnit supports these notations, see e.g. https://github.com/sebastianbergmann/phpunit/pull/6272

`--filter "testSuiteSorterDefectsWithDataProviderTest::testOne#1"`
`--filter "testSuiteSorterDefectsWithDataProviderTest::testOne@data-set"`

meaning, we don't need this `with data set" in the pattern

Reply:
> do I get it right, that we are running

yes. And we do it for PHPUnit <10 and PHPUnit >=10, see

https://github.com/infection/infection/blob/5392be12640e90a3f6f2f3f6a73b1f32b746f562/src/TestFramework/PhpUnit/CommandLine/ArgumentsAndOptionsBuilder.php#L176-L181

if you think we can do it shorter, great! let's extract to a separate issue.

related

- https://github.com/infection/infection/issues/1825#issuecomment-1423104126
- https://github.com/infection/infection/pull/1854
- https://github.com/infection/infection/issues/1899 and the fix https://github.com/infection/infection/pull/1900